File: Solutions by Programming Task/Fibonacci sequence/Java/Recursive.java
import java.math.BigInteger;

public class Main {
    private static BigInteger fib(BigInteger x) {
        return x.compareTo(BigInteger.valueOf(2)) < 0
                ? x
                : fib(x.subtract(BigInteger.valueOf(1))).add(fib(x.subtract(BigInteger.valueOf(2))));
    }

    public static void main(String[] args) {
        System.out.println(fib(BigInteger.valueOf(10)));
    }
}

File: Solutions by Programming Task/Fibonacci sequence/Go/Recursive.go
package main

import "fmt"

func Fib(x int) int {
	if x < 2 {
		return x
	}
	return Fib(x - 1) + Fib(x - 2)
}

func main() {
	fmt.Println(Fib(10))
}
